plot_cfDNA_scATAC/scATAC_demo.Rmd

```diff
@@ -0,0 +1,430 @@
+
+- **Dimension reduction:** We next run singular value decomposition (SVD) on the TD-IDF matrix, using the features (peaks) selected above. This returns a reduced dimension representation of the object (for users who are more familiar with scRNA-seq, you can think of this as analogous to the output of PCA).
+
+
+```{r normalize, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+pbmc <- RunTFIDF(pbmc)
+pbmc <- FindTopFeatures(pbmc, min.cutoff = 'q25')
+pbmc <- RunSVD(pbmc)
+
+# plot the LSI components correlation with depth
+# typically depth is correlated with the first LSI comp.
+
+DepthCor(pbmc)
+
+
+```
+## Make UMAP
+
+```{r umap, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+pbmc <- RunUMAP(object = pbmc, reduction = 'lsi', dims = 2:30)
+pbmc <- FindNeighbors(object = pbmc, reduction = 'lsi', dims = 2:30)
+pbmc <- FindClusters(object = pbmc, verbose = FALSE, algorithm = 3)
+DimPlot(object = pbmc, label = TRUE) + NoLegend()
+```
+
+## Create a gene activity matrix
+
+```{r gene_activity, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# get the activities of the genes from the peaks
+gene.activities <- GeneActivity(pbmc)
+
+# add the gene activity matrix to the Seurat object as a new assay and normalize it
+pbmc[['RNA']] <- CreateAssayObject(counts = gene.activities)
+pbmc <- NormalizeData(
+  object = pbmc,
+  assay = 'RNA',
+  normalization.method = 'LogNormalize',
+  scale.factor = median(pbmc$nCount_RNA)
+)
+
+DefaultAssay(pbmc) <- 'RNA'
+
+DimPlot(object = pbmc, reduction = "umap")
+
+FeaturePlot(
+  object = pbmc,
+  features = c('MS4A1', 'CD3D', 'LEF1', 'NKG7', 'TREM1', 'LYZ'),
+  pt.size = 0.1,
+  max.cutoff = 'q95',
+  ncol = 3
+)
+
+```
+
+
+## now get labels from scRNASeq
+
+
+```{r get_rnaseq, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# Load the pre-processed scRNA-seq data for PBMCs
+expr_file = paste0(proj_dir, "/data/pbmc_10k_v3.rds")
+
+pbmc_rna <- readRDS(expr_file)
+
+transfer.anchors <- FindTransferAnchors(
+  reference = pbmc_rna,
+  query = pbmc,
+  reduction = 'cca'
+)
+
+predicted.labels <- TransferData(
+  anchorset = transfer.anchors,
+  refdata = pbmc_rna$celltype,
+  weight.reduction = pbmc[['lsi']],
+  dims = 2:30
+)
+
+pbmc <- AddMetaData(object = pbmc, metadata = predicted.labels)
+
+```
+
+```{r plot_rnaseq, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+plot1 <- DimPlot(
+  object = pbmc_rna,
+  group.by = 'celltype',
+  label = TRUE,
+  repel = TRUE) + NoLegend() + ggtitle('scRNA-seq')
+
+plot2 <- DimPlot(
+  object = pbmc,
+  group.by = 'predicted.id',
+  label = TRUE,
+  repel = TRUE) + NoLegend() + ggtitle('scATAC-seq')
+
+plot1 + plot2
+
+```
+
+now match clusters from RNA-Seq to ATAC-Seq
+
+```{r match_rnaseq, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+#pbmc <- subset(pbmc, idents = 14, invert = TRUE)
+pbmc <- RenameIdents(
+  object = pbmc,
+  '0' = 'CD14 Mono',
```

Comment:
Will these clusterings always be the same? Is there a chance the ordering will change?

Reply:
no, thankfully they will always be the same as long as nothing in the clustering or the data changes